### DISCUSSION TITLE
How to make it recognize my root path

### DISCUSSION THREAD

**Discussion by eyalk11** [microsoft/pyright#5537] [Q&A]
My root path(c:\\gitproj\\Auto-GPT)  has pyproject.toml and autogpt folder.

On every import, pyright fails to resolve:
```
from autogpt.config import Config  # Import "autogpt.config" could not be resolved Pyright (reportMissingImports) [13, 6]
```
I tried things like : 
```
[tool.pyright]
"executionEnvironments": [
{
  "root": ".",
  "extraPaths": [ "."]

  } ]

```

in pyproject.toml, or 

```
require('lspconfig')['pyright'].setup{
    capabilities = capabilities,
    on_attach = on_attach,
    flags = lsp_flags
    settings = { extraPaths = { 'C:\\gitproj\\Auto-GPT','c:/gitproj/Auto-GPT' } }
}

```
Why is it always soooooo hard? (the 10th times I encounter such things in different products - pyright,pylsp, jedi ) 
Why isn't there a function to list the current included paths? 
And lastly, can you help pls? 
Thanks! 

PS. if I remove the `autogpt.` in the imports it seems to work . But I can't change it project-wise.

**Comment by erictraut** [comment]
When pyright is used in LSP mode, it relies on the client to specify the root directory of the workspace / project. I don't know which client you're using. If you have questions, you can contact the maintainers of your client or the LSP for information on how to properly configure it for use with pyright. Alternatively, you could switch to VS Code where it should just work. ;)

Pyright logs the import resolution paths that it's using. If you're using VS Code, you can go to the "Output" tab and look at the log output. I don't know where the logs are displayed in your client.

**Comment by eyalk11** [comment]
Thanks. I am using vim-lspconfig and I have LspLog. However, I fail to see there anything important. I do see rpc requests and responses.  Can you provide the log location in the configuration? 
Also, is the "." syntax supported? 
My current configuration is 

```
require('lspconfig')['pyright'].setup{
    capabilities = capabilities,
    on_attach = on_attach,
    flags = lsp_flags,
    --root_dir = vim.fs.dirname(vim.fs.find(root_files, { upward = true })[1])
    verboseOutput = true
    --settings = { extraPaths = { 'C:\\gitproj\\Auto-GPT','c:/gitproj/Auto-GPT' } }
}

```
[tool.pyright]
defineConstant = { DEBUG = true }

executionEnvironments = [
  { root = ".", pythonVersion = "3.10", pythonPlatform = "Windows", extraPaths = [ "." ] }
  ]
```
 Client: pyright (id: 5, bufnr: [12])
 	filetypes:       python
 	autostart:       true
 	root directory:  C:/gitproj/Auto-GPT/autogpt
 	cmd:             cmd.exe /C pyright-langserver --stdio

```

I tried to enable debug using 

```
  vim.lsp.set_log_level('debug')
```
I see just RPC. 

I know where the server is located. Not sure where the logs are.

**Reply by erictraut** [reply]
I don't know where vim-lspconfig places the console output of an LSP. You could ask the maintainers. In VS Code, it appears in the "Output" tab.

The "." syntax is supported for configuration settings, but paths that start with "." are relative to the project (workspace) path. It's up to the client to specify the correct workspace path, and that needs to be an absolute path.
